### PR TITLE
Update Hoenn requirements

### PR DESCRIPTION
**Pull request by RedSparr0w**
Update Hoenn Route, Dungeon and Gym requirements.
Update route difficulty order.

List thanks to Ultima:
```
Littleroot Town: none
101: none
Oldale Town: 101
103: 101
102: 101
Petalburg City: 102
104: 102
Petalburg Woods: 104
Rustboro City: Petalburg Woods
116: Petalburg Woods
Rusturf Tunnel: Stone badge
Dewford Town: Rusturf Tunnel
Granite Cave: Rusturf Tunnel
Slateport City: Knuckle badge & Granite Cave
110: Knuckle badge & Granite Cave
Mauville City: 110
117: 110
Verdanturf Town: 117
111: Dynamo badge
112: 111
Fiery Path: 112
113: Fiery Path
Fallarbor Town: 113
114: 113
Meteor Falls: 114
115: Meteor Falls
Mt. Chimney: Meteor Falls
Jagged Pass: Mt. Chimney
Lavaridge Town: Jagged Pass
Petalburg City Gym: Heat badge
105: Balance badge
106: 105/107
107: 106/108
108: 107/109
109: Balance badge
New Mauville: Balance badge
118: Balance badge
119: 118
Fortree City: 119
120: 119
121: 120
Lilycove City: 121
122: 121 & Feather badge
Mt. Pyre: 122
123: 122
124: Mt. Pyre
125: 124
Shoal Cave: 125
Mossdeep City: 125
126: 124
127: 125/126
128: 127
129: 128
130: 129
131: 130
Pacifidlog Town: 131
132: 131
133: 132
134: 133
Sootopolis City: Mind Badge & 126
Seaflor Cavern: Mind Badge & 128
Cave of Origin: Seafloor Cavern & 126
Sky Pillar: Cave of Origin & 131
Sootopolis City Gym: Sky Pillar
Ever Grande City: Rain Badge
Victory Road Hoenn: Rain Badge
Pokemon League Hoenn: Victory Road Hoenn

Route difficulty order: 101, 103, 102, 104, 116, 110, 117, 111, 112, 113, 114, 115, 105, 106, 107, 108, 109, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134
```

### CODE REVIEW

File: src/scripts/towns/Town.ts [hunk 1]
```diff
@@ -215,34 +215,34 @@ const BattleFrontierShop = new Shop([
...
-TownList['Sootopolis City'] = new Town('Sootopolis City', [new RouteKillRequirement(10, 126), new ClearDungeonRequirement(1, Statistics.getDungeonIndex('Cave of Origin'))]);
-TownList['Ever Grande City'] = new Town('Ever Grande City', [new RouteKillRequirement(10, 128)]);
+TownList['LilyCove City'] = new Town('LilyCove City', [new RouteKillRequirement(10, 121)], LilyCoveCityShop);
+TownList['Mossdeep City'] = new Town('Mossdeep City', [new RouteKillRequirement(10, 125)], MossdeepCityShop);
+TownList['Sootopolis City'] = new Town('Sootopolis City', [new RouteKillRequirement(10, 126), new GymBadgeRequirement(BadgeCase.Badge.Mind)]);
+TownList['Ever Grande City'] = new Town('Ever Grande City', [new GymBadgeRequirement(BadgeCase.Badge.Rain)]);
 TownList['Pokemon League Hoenn'] = new Town('Pokemon League', [new RouteKillRequirement(10, 128), new ClearDungeonRequirement(1, Statistics.getDungeonIndex('Victory Road Hoenn'))]);
 TownList['Pacifidlog Town'] = new Town('Pacifidlog Town', [new RouteKillRequirement(10, 131)]);
 TownList['Battle Frontier'] = new Town('Battle Frontier', [new ClearDungeonRequirement(1, Statistics.getDungeonIndex('Victory Road Hoenn'))], BattleFrontierShop);
```
Comment (apple096) [inline src/scripts/towns/Town.ts]: This is not right, Frontier is post-League

**Review by apple096 — CHANGES_REQUESTED** [review]

File: src/scripts/towns/Town.ts (hunk 1 above)
Comment (RedSparr0w) [reply src/scripts/towns/Town.ts]:
```suggestion
TownList['Battle Frontier'] = new Town('Battle Frontier', [new GymBadgeRequirement(BadgeCase.Badge.Elite_HoennChampion)], BattleFrontierShop);
```

**Review by apple096 — APPROVED** [review]
Nice!

**Review by Aegyo — APPROVED** [review]
:+1:

**Pull request merged**